lib/scholar/options.ex

```diff
@@ -30,4 +32,32 @@ defmodule Scholar.Options do
       {:error, "expected a non-negative number, got: #{inspect(num)}"}
     end
   end
+
+  def weights(weights) do
+    if Nx.is_tensor(weights) or is_list(weights) do
+      {:ok, weights}
+    else
+      {:error, "expected weights to be a tensor or a list, got: #{inspect(weights)}"}
+    end
+  end
+
+  def positive_weights(weights) do
+    if (Nx.is_tensor(weights) and Nx.to_number(Nx.all(Nx.greater(weights, 0.0))) == 1) or
+         (is_list(weights) and Enum.all?(weights, fn x -> x > 0.0 end)) do
+      {:ok, weights}
+    else
+      {:error,
+       "expected weights to be a tensor or a list of positive numbers, got: #{inspect(weights)}"}
+    end
+  end
+
+  def non_negative_weights(weights) do
+    if (Nx.is_tensor(weights) and Nx.to_number(Nx.all(Nx.greater_equal(weights, 0.0))) == 1) or
+         (is_list(weights) and Enum.all?(weights, fn x -> x >= 0.0 end)) do
```

Comment:
I wonder if x being a list would compare as bigger than a float. I'm almost sure it would